Source/Serilog.Exceptions/Core/ExceptionPropertiesBag.cs

```diff
@@ -56,10 +56,45 @@ public void AddProperty(string key, object? value)
                 }
             }
 
-            this.properties.Add(key, value);
+            this.AddPairToProperties(key, value);
         }
 
         /// <inheritdoc />
         public bool ContainsProperty(string key) => this.properties.ContainsKey(key);
+
+        private void AddPairToProperties(string key, object? value)
+        {
+#if NET5_0
+            var i = 0;
+            while (!this.properties.TryAdd(key, value) && i < 5)
+            {
+                key += "$";
+                i++;
+            }
+#else
+            key = this.MakeSureKeyIsUnique(key);
+
+            this.properties.Add(key, value);
+#endif
+        }
+
+#if !NET5_0
+        /// <summary>
+        /// We want to be as robust as possible
+        /// so even in case of multiple properties with the same name
+        /// we want to at least try carrying on and keep working.
+        /// </summary>
+        private string MakeSureKeyIsUnique(string key)
+        {
+            var i = 0;
+            while (this.properties.ContainsKey(key) && i < 5)

```

Comment:
Won't this still silently fail if there are 5 properties with same name?

Reply:
Yes, it still will :-).
Why I did it that way?
a) I would never leave an infinite loop in a production code. Even though it is simple code in our case I am still terrified of infinite loops :-).
b) This is just a "defensive" code mechanism that happens to also "fix" #343. After that is merged I will prepare a second PR where we will actually be taking into account the "new" modifier in restructured classes.
c) After few years of library existence, it is the first time when someone reported a problem with exceptions with redefined properties(1 level of "new" modifiers). I assume 5 levels of "new" modifiers is too crazy for real-world and expect this code to work just fine in practice.